contracts/JetStaking/JetStakingV3.sol

```diff
@@ -0,0 +1,172 @@
+    // mixing old weighted shares stream calculations and the new ones.
+    mapping(uint256 => bool) public oldWeightedShareStreams;
+
+    /// @dev set the old weighted share streams
+    /// @notice This function must be called prior upgrading the contract.
+    /// And the contract must be paused before initializing it.
+    function initializeOldWeightedStreams()
+        public
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        require(paused != 0, "E16"); // REQUIRE_PAUSE
+        for (uint256 i = 1; i < streams.length; i++) {
+            oldWeightedShareStreams[i] = true;
+        }
+    }
+
+    /// @dev calculates and gets the latest reward per share (RPS) for a stream
+    /// @param streamId stream index
+    /// @return streams[streamId].rps + scheduled reward up till now
+    function getLatestRewardPerShare(uint256 streamId)
+        public
+        view
+        override
+        returns (uint256)
+    {
+        require(streamId != 0, "E29"); // AURORA_REWARDS_COMPOUND
+        uint256 totalShares = oldWeightedShareStreams[streamId]
+            ? totalStreamShares
+            : totalAuroraShares;
+        require(totalShares != 0, "E30"); // ZERO_STREAM_SHARES
+        return
+            streams[streamId].rps +
+            (getRewardsAmount(streamId, touchedAt) * RPS_MULTIPLIER) /
+            totalShares;
+    }
+
+    /// @dev gets the user's stream claimable amount
+    /// @param streamId stream index
+    /// @return (latesRPS - user.rpsDuringLastClaim) * user.shares
+    function getStreamClaimableAmount(uint256 streamId, address account)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        uint256 latestRps = getLatestRewardPerShare(streamId);
+        User storage userAccount = users[account];
+        uint256 userRps = userAccount.rpsDuringLastClaim[streamId];
+        uint256 userShares = oldWeightedShareStreams[streamId]
+            ? userAccount.streamShares
+            : userAccount.auroraShares;
+        return ((latestRps - userRps) * userShares) / RPS_MULTIPLIER;
+    }
+
+    /// @dev gets a user stream shares
+    /// @param streamId stream index
+    /// @param account the user address
+    /// @return user stream shares
+    function getAmountOfShares(uint256 streamId, address account)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        uint256 userShares = (oldWeightedShareStreams[streamId] &&
+            streamId != 0)
+            ? users[account].streamShares
+            : users[account].auroraShares;
+        return userShares;
+    }
+
+    /// @dev calculate the total amount of the released tokens within a period (start & end)
+    /// @param streamId the stream index
+    /// @param start is the start timestamp within the schedule
+    /// @param end is the end timestamp (e.g block.timestamp .. now)
+    /// @return amount of the released tokens for that period
+    function rewardsSchedule(
```

Comment:
This function seems same as `v2` contract, we don't need to override in this case. Please let me know if I am missing something!